eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -114,11 +114,10 @@ stages:
                     PackageName: ${{artifact.name}}
                     ServiceName: ${{parameters.ServiceDirectory}}
                     ForRelease: true
-
-          - template: /eng/pipelines/templates/jobs/installation.check.yml
-            parameters:
-              Artifact: ${{ artifact }}
-              ArtifactName: ${{ parameters.ArtifactName }}
+              - template: /eng/pipelines/templates/steps/compilation.check.yml
```

Comment:
While the maven goal is `compile`, I think the `compile` terminology is somewhat misleading and maybe a more descriptive name for the yaml file would be good. Unless it's clear enough to anyone in the java ecosystem that it's fine? @alzimmermsft